Makefile

```diff
@@ -13,18 +13,24 @@
 
 # Set this to pass additional commandline flags to the go compiler, e.g. "make test EXTRAGOARGS=-v"
 EXTRAGOARGS:=
+DISABLE_ROOT_TESTS?=1
 
 all: build
 
 test: all-tests
 
-unit-tests:
-	go test -short ./... $(EXTRAGOARGS)
+unit-tests: test-setup
+	DISABLE_ROOT_TESTS=$(DISABLE_ROOT_TESTS) go test -short ./... $(EXTRAGOARGS)
 
-all-tests:
-	go test ./... $(EXTRAGOARGS)
+all-tests: test-setup
+	DISABLE_ROOT_TESTS=$(DISABLE_ROOT_TESTS) go test ./... $(EXTRAGOARGS)
+
+test-setup:
```

Comment:
You can have make track the files so it knows whether or not to run this target each time.  Something like this:

```make
testdata_objects = testdata/vmlinux testdata/root-drive.img testdata/firecracker
test-setup: $(testdata_objects)
	@test -w /dev/kvm || \
		(echo "In order to run firecracker, $(shell whoami) must have write permission to /dev/kvm"; false)

$(testdata_objects):
	./prepare-testdata.sh
```

Reply:
If we use make to track the files, we need to pass the target to the shell script to only download the missing file.

```
$(testdata_objects):
	./prepare-testdata.sh $@
```

Another path we could take is having each target separately.

```
testdata/vmlinux:
        curl -fsSL -o $@ https://s3.amazonaws.com/spec.ccfc.min/img/hello/kernel/hello-vmlinux.bin

```

Reply:
Removed the shell script and used `make` for tracking existing files on https://github.com/firecracker-microvm/firecracker-go-sdk/pull/144/commits/c210e9bf1f8a2da501c21fe240463ab8b23bb3c8